readdy/main/model/observables/ReactionCounts.cpp

```diff
@@ -45,22 +45,27 @@ namespace observables {
 struct ReactionCounts::Impl {
     using data_set_t = io::DataSet<std::size_t, false>;
     std::unique_ptr<io::Group> group;
-    std::unique_ptr<data_set_t> ds_order1;
-    std::unique_ptr<data_set_t> ds_order2;
+    std::unordered_map<readdy::model::Particle::type_type, std::unique_ptr<data_set_t>> ds_order1;
```

Comment:
i think a map type -> data_set suffices, the unique_ptr is not really needed anymore (it was just so it hadn't to be initialized).